stretch_core/stretch_core/stretch_driver.py

```diff
@@ -108,6 +112,62 @@ def set_mobile_base_velocity_callback(self, twist):
         self.angular_velocity_radps = twist.angular.z
         self.last_twist_time = self.get_clock().now()
         self.robot_mode_rwlock.release_read()
+    
+    def set_robot_streaming_position_callback(self, msg):
+        self.robot_mode_rwlock.acquire_read()
+        if not self.streaming_position_activated:
+            self.get_logger().error('Streaming position is not activated.'
+                                    ' Please activate streaming position to receive command to joint_position_cmd.')
+            self.robot_mode_rwlock.release_read()
+            return
+        
+        if not self.robot_mode in ['position', 'navigation']:
+            self.get_logger().error('{0} must be in position or navigation mode with streaming_position activated ' 
+                                    'enabled to receive command to joint_position_cmd. '
+                                    'Current mode = {1}.'.format(self.node_name, self.robot_mode))
+            self.robot_mode_rwlock.release_read()
+            return
+
+        
+        if self.get_clock().now() - self.streaming_controller_lt.last_update_time> 5:
+            if STREAMING_POSITION_DEBUG:
+                print('Reset Streaming position looptimer after 5s no message received.')
+            self.streaming_controller_lt.reset()
+        qpos = msg.data
+        self.move_to_position(qpos)
+        self.robot_mode_rwlock.release_read()
+        self.streaming_controller_lt.update()
+    
+    def move_to_position(self, qpos):
```

Comment:
I know we don't do that in this repo but I still think it would be good to move towards type hint + docstring for things like this

what dimensiionality should qpos be for example